sdk/core/azure-core/src/http/curl/curl.cpp

```diff
@@ -75,13 +80,29 @@ int pollSocketUntilEventOrTimeout(
   // Call poll with the poller struct. Poll can handle multiple file descriptors by making an pollfd
   // array and passing the size of it as the second arg. Since we are only passing one fd, we use 1
   // as arg.
+
+  // Cancelation is possible by calling poll() with small time intervals instead of using the
```

Comment:
This timeout needs to be dialable.  There is a lot of busy wait CPU spinning that will occur.